apps/api/src/controllers/applicationController.ts

```diff
@@ -59,22 +60,49 @@ export const editApplication = async ({ id, update }: { id: number; update: Appl
 		return result;
 	}
 
-	const { state } = result.data;
+	const application = result.data;
 
-	// TODO: Replace w/ state machine https://github.com/Pan-Canadian-Genome-Library/daco/issues/58
-	const isEditState =
-		state === ApplicationStates.DRAFT ||
-		state === ApplicationStates.INSTITUTIONAL_REP_REVIEW ||
-		state === ApplicationStates.DAC_REVIEW;
+	const { edit } = ApplicationStateEvents;
+	const canEditResult = new ApplicationStateManager(application)._canPerformAction(edit);
 
-	if (isEditState) {
-		const result = await applicationRepo.editApplication({ id, update });
-		return result;
-	} else {
-		const message = `Cannot update application with state ${state}`;
+	if (!canEditResult.success) {
+		const message = `Cannot update application with state ${application.state}`;
 		logger.error(message);
 		return failure(message);
 	}
+
+	const formatedUpdate: ApplicationContentUpdates = {
```

Comment:
I would make this a reusable function in case this scenario pops up again
The reverse utility exists here: https://github.com/Pan-Canadian-Genome-Library/daco/blob/main/apps/api/src/utils/routes.ts#L40

Also variable name should be `formattedUpdate` (missing 't')